lib/Room.php

```diff
@@ -633,6 +633,15 @@ public function changeInCall($sessionId, $active) {
 	 * @return bool
 	 */
 	public function verifyPassword($password) {
+		$event = new GenericEvent(null, [
+			'password' => $password,
+			'hasPassword' => $this->hasPassword(),
+		]);
```

Comment:
You should pass the room as subject of the event, so the event handler knows which room is checking the password and has access to all parameters:
```php
new GenericEvent($this, [
    'password' => $password,
]);
```